### DISCUSSION TITLE
How automatically add a reviewer from a team when a PR is created?

### DISCUSSION THREAD

**Discussion by Tom-Boscher** [community/community#47988] [New to GitHub] [answered]
Hello everyone!

We recently move from Gitlab to GitHub and I struggle with a automatic reviewer assignment.
My goal is to automatically add a team `QA` to the reviewers when a PR is created and to only be able to merge the PR if one of the `QA` reviewer has validated the review.

Thanks in advance!

**Comment by spenserblack** [comment, marked as answer]
It sounds like a [`CODEOWNERS` file](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners) is what you need for automatic reviewer assignments :slightly_smiling_face:

You can add a [branch protection rule](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/about-protected-branches#require-pull-request-reviews-before-merging) to require reviews before merging.

**Reply by joeythomaschaske** [reply]
I want to separate the "team" from the "owner" so that I can require approval from the "owner" when the "team" submit pull requests.

Team members can review PRs, but I want final approval to come from the "Owner"

**Comment by Tom-Boscher** [comment]
Thank you @spenserblack for your answer! It seems indeed to be the solution for my issue.
Is it possible to assign the team only when a PR is reviewed?

**Reply by spenserblack** [reply]
> Is it possible to assign the team only when a PR is reviewed?

:thinking: If I'm understanding correctly, you want to assign a team to review after the PR has been reviewed, like a 2-stage review process? If so, you can [trigger a workflow on approval](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#running-a-workflow-when-a-pull-request-is-approved), then use the GitHub CLI (`gh pr edit --add-reviewer foo`).

**Reply by Tom-Boscher** [reply]
Thank you, I will try this way so!

**Reply by spenserblack** [reply]
I happened to stumble on [this action](https://github.com/wow-actions/auto-assign), which may interest you.

**Reply by Tom-Boscher** [reply]
Oh indeed, thank you!

**Comment by jeganila** [comment]
I have the githup/CODEOWNERS with below content but when i create pull request , it is not assigning the default reviewers .
branch rule is setup already 

```# This is a comment.
# Each line is a file pattern followed by one or more owners.
# These owners will be the default owners for everything in
# the repo. Unless a later match takes precedence,
* @it-devops/it-devops

**Reply by spenserblack** [reply]
For organization teams the format should be `@org/team`. See https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners#codeowners-syntax

Are you using a self-hosted GitHub instance? Because when I look up `@it-devops`, I find a user instead of an organization.
